src/appleseed.studio/mainwindow/rendering/renderwidget.cpp

```diff
@@ -422,5 +428,29 @@ void RenderWidget::paintEvent(QPaintEvent* event)
     m_painter.end();
 }
 
+void RenderWidget::dragEnterEvent(QDragEnterEvent* event)
+{
+    if (event->mimeData()->hasFormat("text/plain") && m_project.has_trace_context())
+        event->acceptProposedAction();
+}
+
+void RenderWidget::dragMoveEvent(QDragMoveEvent* event)
+{
+    if (event->pos().x() > pos().x() + width() || event->pos().y() > pos().y() + height())
```

Comment:
Three questions:

- What are these checks for exactly?
- Are you sure it shouldn't be `pos().x() + width() - 1`? Or, alternatively, `>=` instead of `>`?
- Isn't checking the other sides (< 0) also necessary?

Reply:
This checks are to be sure the user can only drop the material inside the scene. The default behavior always start when you enter the area but have not the behaviour wanted when it leaves the area, so this "if" is here to fix this.
I just checked, and yes, it has an undefined behavior when you leaves the area by the top or left side so I will add more checks.